solidity/deploy/21_genesis_relay.ts

```diff
@@ -0,0 +1,31 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+import { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { getNamedAccounts, deployments } = hre
+  const { execute } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  // The genesis header comes form the Bitcoin mainnet block at height 766080
+  // which is the first block of Bitcoin difficulty epoch 380.
+  const genesisHeader =
+    "0x0000402089138e40cd8b4832beb8013bc80b1425c8bcbe10fc28040000000000000000" +
+    "0058a06ab0edc5653a6ab78490675a954f8d8b4d4f131728dcf965cd0022a02cdde59f8e" +
+    "63303808176bbe3919"
+  const genesisHeight = 766080
+  const genesisProofLength = 20
+
+  await execute(
+    "LightRelay",
+    { from: deployer, log: true, waitConfirmations: 1 },
+    "genesis",
+    genesisHeader,
+    genesisHeight,
+    genesisProofLength
+  )
+}
+
+export default func
+
+func.tags = ["GenesisLightRelay"]
+func.dependencies = ["LightRelay"]
```

Comment:
We should use `func.skip` property to determine networks for which scripts from this PR should be executed.
See the example here: https://github.com/keep-network/tbtc-v2/blob/6ff3d7b1c7d0a421507037acf3a421c4261dfcdd/solidity/deploy/90_initialize_wallet_owner.ts#L20-L21

I guess we don't want to run them for `goerli` as we use `TestRelay` stub instead of the `LightRelay` there:
https://github.com/keep-network/tbtc-v2/blob/6ff3d7b1c7d0a421507037acf3a421c4261dfcdd/solidity/deploy/00_resolve_relay.ts#L21-L22

Reply:
Indeed. 
In ed8983798ba6d06892072267ad970e9263fbc254 I added to all the scripts that modify `LightRelay`: 
```
func.skip = async (hre: HardhatRuntimeEnvironment): Promise<boolean> =>
  hre.network.name === "goerli"
 ```

Reply:
I think we should rather use:
```
hre.network.name !== "mainnet"
```

We not only want to skip this step for Goreli but also for local deployment that also uses a stub. This step is only for mainnet.

Reply:
Done in d13fe6ff3af383afd56e3fa7c13b6fef8f2c252f.